### DISCUSSION TITLE
How to pass variable to Stats overview and chart widgets

### DISCUSSION THREAD

**Discussion by valpuia** [filamentphp/filament#6063] [Help] [answered]
I want to display stats overview or chart inside detail (custom) page, but I need to check data base on that particular `$id`

Any suggestion to pass ID to the widgets and access that ID in another class

```
// My Custom Detail Page
protected function getFooterWidgets(): array
{
    return [
        ExchangeOrderChart::class
    ];
}
```

```
// ChartWidget
class ExchangeOrderChart extends LineChartWidget
{
    // how to access ID inside this widget class
    protected static ?string $heading = 'Chart';

    public ?Model $record = null;

    protected function getData(): array
    {
        dd($this->record); // null
        return [];
    }
}
```

This line always give me null
[https://filamentphp.com/docs/2.x/admin/resources/widgets#accessing-the-current-record](https://filamentphp.com/docs/2.x/admin/resources/widgets#accessing-the-current-record)

**Comment by danharrin** [comment, marked as answer]
In your page view, pass in any property data with an array:

```blade
<x-filament::page :widget-data="['record' => $record]">
```

**Reply by TegarAditya** [reply]
I tried this on my custom page but it shows this error:

`Undefined variable $record`

![image](https://github.com/filamentphp/filament/assets/71318821/a2e210fb-4e56-4f21-a25b-ed45ff16b78e)

This error appears even though I have used the {record} parameter in the url in the resource class

```
//In the SurveyResource.php

...
public static function getPages(): array
    {
        return [
            'index' => Pages\ListSurveys::route('/'),
            'create' => Pages\CreateSurvey::route('/create'),
            'edit' => Pages\EditSurvey::route('/{record}/edit'),
            'view' => Pages\ViewSurvey::route('/{record}'),
            'result' => Pages\SurveyResult::route('/{record}/result'),
        ];
    }
...
```

```
// In my SurveyResult.php (the custom pages)

class SurveyResult extends Page
{
    protected static string $resource = SurveyResource::class;

    protected static string $view = 'filament.resources.survey-resource.pages.survey-result';
    
    ...
}
```

Any suggestions to resolve the error?

Thank You,

**Reply by valpuia** [reply]
https://discord.com/channels/883083792112300104/1116265881547321394 check this link

**Reply by valpuia** [reply]
Or declare variable $record or $abc in SurveyResult.php and check in view

**Reply by TegarAditya** [reply]
First, I need to declare $record in the custom page class then mount it with the model

```
//In the SurveyResult.php

...

public $records;

public function mount()
{
    $this->record = Survey::findOrFail(request()->route('record'));
}

...
```

Thank you for the suggestion. That fixed the problem right away!

**Reply by valpuia** [reply]
You can access variable inside mount()

```
public $abc;

public function mount($record)
{
    $this->abc = Survey::findOrFail($record);
}
```

**Reply by TegarAditya** [reply]
The code you provided works very well. I tested it, and everything functions as expected. Thank you for sharing.
